amy/dashboard/views.py

```diff
@@ -238,25 +269,33 @@ def training_progress(request):
         get_involved_form = GetInvolvedForm(
             data=request.POST, instance=base_training_progress
         )
+
         if get_involved_form.is_valid():
-            get_involved_form.save()
+            return self.form_valid(get_involved_form)
 
-            messages.success(
-                request,
-                "Thank you. Your Get Involved submission will be reviewed within 7-10 "
-                "days.",
-            )
-            return redirect(reverse("training-progress"))
+        else:
+            return self.form_invalid(get_involved_form)
 
-    context = {
-        "title": "Your training progress",
-        "get_involved_form": get_involved_form,
-        "progress_training": progress_training,
-        "progress_get_involved": progress_get_involved,
-        "progress_welcome": progress_welcome,
-        "progress_demo": progress_demo,
-    }
-    return render(request, "dashboard/training_progress.html", context)
+
+class GetInvolvedUpdateView(LoginRequiredMixin, RedirectSupportMixin, AMYUpdateView):
+    # permission_required = TODO
+    model = TrainingProgress
+    form_class = GetInvolvedForm
+    template_name = "get_involved_form.html"
+    success_url = reverse_lazy("training-progress")
+    success_message = "Your Get Involved submission was updated successfully."
+
+    def get_context_data(self, **kwargs) -> dict:
+        context = super().get_context_data(**kwargs)
+        context["title"] = str("Update your Get Involved submission")
+        return context
+
+
+class GetInvolvedDeleteView(LoginRequiredMixin, RedirectSupportMixin, AMYDeleteView):
```

Comment:
`RedirectSupportMixin` not needed here, as `AMYDeleteView` handles redirection to `success_url` on its own.